containers/src/Data/Set/Internal.hs

```diff
@@ -897,6 +905,24 @@ intersection t1@(Bin _ x l1 r1) t2
 {-# INLINABLE intersection #-}
 #endif
 
+#if (MIN_VERSION_base(4,9,0))
+-- | The intersection of a series of sets. Intersections are performed left-to-right.
+intersections :: Ord a => NonEmpty (Set a) -> Set a
+intersections (s0 :| ss) = List.foldr go id ss s0
+    where
+      go s r acc
+          | null acc = empty
+          | otherwise = r (intersection acc s)
+
+-- | Sets form a 'Semigroup' under 'intersection'.
```

Comment:
The docs should mention that this type is only included for `base >= 4.9`.

Reply:
Better, I think: include the type and let it be useless without the `Semigroup` instance.